compiler/backend/cbackend.nim

```diff
@@ -67,6 +67,10 @@ import
 
 import std/options as std_options
 
+from compiler/front/msgs import localReport
+from compiler/ast/reports import ReportKind
+from compiler/ast/reports_sem import SemReport
```

Comment:
Here I'd note report code as tech debt, especially given that we're producing a "sem" error from the backend.

Reply:
I'll add a note here, but `rsemHintLibDependency` is not an error and also not really something related to sem -- it's a trace at most.

Reply:
Ah, I zoned on the `hint`, I see "report" and it's usually an error. But very much agree that it's all very weird.

Reply:
I added a note both here and where `localReport` is used.